firefox-ios/Client/Frontend/Settings/AppearanceSettings/AppearanceSettingsView.swift

```diff
@@ -126,6 +150,28 @@ struct AppearanceSettingsView: View, FeatureFlaggable {
         }
     }
 
+    private struct NavigationToolbarSection: View {
+        let theme: Theme?
+        let selectedOption: NavigationBarMiddleButtonType
+        let onChange: (NavigationBarMiddleButtonType) -> Void
+        let cornerRadius: CGFloat
+
+        var body: some View {
+            GenericSectionView(
+                theme: theme,
+                title: .Settings.Appearance.NavigationToolbar.SectionHeader,
+                description: .Settings.Appearance.NavigationToolbar.SectionDescription,
+                identifier: AccessibilityIdentifiers.Settings.Appearance.navigationToolbarSectionTitle
+            ) {
+                NavigationBarMiddleButtonSelectionView(
+                    theme: theme,
+                    selectedMiddleButton: selectedOption,
+                    onSelected: onChange)
```

Comment:
curious on why we need to pass in `selectedOption` and `onChange` when its reference in the same file. Maybe we can keep it self contained for `NavigationToolbarSection`. 

Would be nice to maybe to add `// MARK: NavigationToolbarSection` and add all related code under it?